File: php/features.php
<?php
declare(strict_types=1);

// NdbcBuoyData SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class NdbcBuoyDataFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new NdbcBuoyDataBaseFeature();
            case "test":
                return new NdbcBuoyDataTestFeature();
            default:
                return new NdbcBuoyDataBaseFeature();
        }
    }
}

File: py/features.py
# NdbcBuoyData SDK feature factory

from feature.base_feature import NdbcBuoyDataBaseFeature
from feature.test_feature import NdbcBuoyDataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NdbcBuoyDataBaseFeature(),
        "test": lambda: NdbcBuoyDataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
